### PR TITLE
Feature/phenx parser

### PR DESCRIPTION
**Pull request by YaphetKG**
* Parses Phenx protocol xlsx file
* introduces linked_elements array in DugElement data class, to maintain linkage between items in same row. (this will be used by roger to extract and build KG)
* Parses Ontolgical terms in spread sheet as Dug COncepts and attaches them to LOINC , PHENX and CDE elements in that same row. The ontological terms are also normalized.
* Tests these features.

### CODE REVIEW

File: src/dug/core/parsers/phenx_protocol_parser.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        cde_elements = []
+        # Loinc elements from all rows
+        loinc_elements = []
+        # phenx elements from all rows
+        phenx_elements = []
+        # CDE cols from all rows  with ontology ids
+        other_concepts = []
+
+        # For each row
+        for row in PhenXParser.get_parsed_row(input_file):
```
Comment (cschreep) [inline src/dug/core/parsers/phenx_protocol_parser.py:57]: Everything in the body of this for-loop should be moved to a separate function

File: src/dug/core/parsers/phenx_protocol_parser.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            for col_name in cols:  # , col_id in col_pairs:
+                # col_to_curie_prefix_map = {'PhenX': 'PhenX', 'LOINC': 'LOINC', 'CDE': 'CDE'}
+                if pd.notna(row[1][col_name]):
+                    row_dict[col_name] = str(row[1][col_name]).strip()
+            yield row_dict
+
+    def __call__(self, input_file: InputFile)-> List[Indexable]:
+        config = DugConfig.from_env()
+        factory = DugFactory(config)
+        http_session = factory.build_http_session()
```
Comment (cschreep) [inline src/dug/core/parsers/phenx_protocol_parser.py:44]: The declaration of this element is far removed from its use

**Review by cschreep — CHANGES_REQUESTED** [review]
Some of the code in the body of for-loops or conditionals can be extracted into testable functions

**Comment by YaphetKG** [comment]
Instead of parsing Phenx csv here, we will adopt the way the pattern of parsing it to a common format that would be used by one of the dug parsers

**Pull request closed without merging**